src/schema/posts.ts

```diff
@@ -1622,19 +1626,22 @@ export const resolvers: IResolvers<any, Context> = {
         id,
         commentary,
         sourceId,
-      }: { id: string; commentary: string; sourceId: string },
+        postId,
+      }: { id: string; commentary: string; sourceId: string; postId?: string },
       ctx,
       info,
     ): Promise<GQLPost> => {
       await ctx.con.getRepository(Post).findOneByOrFail({ id });
       await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
 
-      const newPost = await createSharePost(
+      const newPost = await upsertSharePost(
         ctx.con,
         sourceId,
         ctx.userId,
         id,
         commentary,
+        true,
+        postId,
```

Comment:
lack of named arguments usage results in this, I can refactor this further if we want to clean it up

Reply:
In API project we don't use named arguments most of the times. I am up for it to start but maybe see what @idoshamun thinks.

Reply:
It's for legacy reasons, I'm up to using them. It's not a guideline :)

Reply:
I prefer named as well, I think for new/refactored stuff we can start implementing it (if scope stays small)